### DISCUSSION TITLE
Unable to deploy Next.js application to Vercel

### DISCUSSION THREAD

**Discussion by lukebettridge** [vercel/next.js#25011] [Help] [answered]
Good evening. Unfortunately I've been pulling my hair out trying to get a basic Next app deployed to Vercel. My application runs and builds fine locally, but returns the following error when attempting to build and deploy to Vercel:

```
info  - Compiled successfully
info  - Collecting page data...
Warning: You have opted-out of Automatic Static Optimization due to `getInitialProps` in `pages/_app`. This does not opt-out pages with `getStaticProps`
Read more: https://nextjs.org/docs/messages/opt-out-auto-static-optimization
> Build error occurred
Error: Failed to load /portfolio
    at loadComponents (/vercel/path0/node_modules/next/dist/next-server/server/load-components.js:1:1554)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async /vercel/path0/node_modules/next/dist/build/utils.js:23:2348
    at async Span.traceAsyncFn (/vercel/path0/node_modules/next/dist/telemetry/trace/trace.js:6:584) {
  type: 'Error'
}
```

`/portfolio` being one of my pages.

I am using `graphql-request` to fetch data from GraphCMS to generate static paths and props within `getStaticPaths` and `getStaticProps`. You may find the application [here](https://github.com/lukebettridge/hartley-charlton).

I have also checked a dozen times, my environment variables _are_ set correctly within the Vercel console.

**Comment by bobarros** [comment]
When you say it's running fine locally, you're talking about 'next build' and then 'next start?' 

I can't test your repository, since the env variables are not available. It will be too much work.

One thing that I have an opinion on is: I would never reach for getInitialProps only to have a Dark Theme.

**Reply by lukebettridge** [reply]
Hi @bobarros, thanks for taking the time. That's correct, using `next build` and `next start`. I've made the GraphCMS endpoint public and removed the need for environment variables on [this](https://github.com/lukebettridge/hartley-charlton/tree/remove-graphcms-variables) branch.

You're quite right, I've only just noticed the HOC is adding `getInitialProps`.

**Comment by lukebettridge** [comment, marked as answer]
After downgrading from `next@10.2.0` to `next@10.0.9`, the error messages within the build logs became less obscure leading me to discover that I was missing a peer-dependency. All is now working as expected.

**Reply by OkkarMin** [reply]
I can attest to this, the missing module for me was @babel/core,

all I had to do was

`npm install @babel/core`
